R/item_upload_files.R

```diff
@@ -120,3 +120,145 @@ multi_file_body <- function(files){
 	names(body) = rep('file', length(body))
 	return(body)
 }
+
+#' @title Upload File to Item Cloud Storage
+#' @description Adds a file to an item in cloud storage
+#'
+#' @template manipulate_item
+#' @inheritParams item_upload_create
+#' @param status logical display upload status?
+#'
+#' @return Success message invisibly. NOTE: cloud processing
+#' can take some time so the added file may not appear immediately. 
+#' For this reason, a sciencebase item json is NOT returned as is
+#' done with other similar functions.
+#'
+#' @import httr
+#' @importFrom mime guess_type
+#'
+#' @examples \dontrun{
+#' res <- item_create(user_id(), "testing 123")
+#' cat("foo bar", file = "foobar.txt")
+#' item_append_files(res$id, "foobar.txt")
+#' }
+#' @export
+item_upload_cloud <- function(sb_id, files, status = TRUE, session=current_session()) {
+
+	try(sb_id <- sb_id$id, silent = TRUE)
+	
+	for(file in files) {
+		if(status)
+			message(paste("Uploading:", file))
+		
+		mimetype <- guess_type(file, unknown = 'application/octet-stream')
+		
+		cloud_upload(file, mimetype, sb_id)
+	}
+	
+	return(invisible("Success"))
+	
+}
```

Comment:
I'm looping for multiple files here the way we do for item_append_files, that seem ok?

Reply:
Yeah, wouldn't be much use in paralellizing